### ISSUE TITLE
Starting the NYT news chatbot

### ISSUE THREAD

**Issue by scripting** [scripting/Scripting-News#256]
Let's learn how to start and update a chatthing.ai bot. Note I am not reporting any problems in this, it all worked. I'm just "narrating my work."

- I created a new bot called NYT headlines.
--Here's a <a href="http://scripting.com/images/2023/04/07/chatthingscreen1.png">screen shot</a>. 
--I'm going to upload files using <a href="http://scripting.com/images/2023/04/07/chatthingscreen2.png">this interface</a>.
--To start the experiment, I'm going to create a Markdown-ized RSS feed from the NYT.
- Let's start with their <a href="http://nytimes.com/timeswire/feeds/">Most Recent Headlines</a> feed. 
- Here's the <a href="http://feeder.scripting.com/returnmarkdown?feedurl=http://nytimes.com/timeswire/feeds/">Markdown-ized version</a> of that feed. 
- Here's the <a href="http://scripting.com/publicfolder/chatgpt/nytBotSource/mostRecentHeadlines.md">resulting file</a> that we'll give to chatthing to ingest. 
- I loaded the file it created into the chatthing bot, and then asked a question: "What's new from the NYT?"
- This is [what I got back](https://chatthing.ai/bots/8150dc18-2086-428d-81a1-946248c381a8/chats/cb757963-6a89-4034-9410-ac488b8f21ea)...
- In other words -- it worked. The formatting sucks, I would have preferred a bulleted list, and it left out the time stamps. But I got some news. Now to figure out how to get it to read this file periodically.

**Comment by scripting** [comment]
Next up --

Add more feeds.

**Comment by scripting** [comment]
The NYT newsbot now knows about NBA news.

https://chatthing.ai/bots/8150dc18-2086-428d-81a1-946248c381a8

**Comment by drewkime** [comment]
Can you just ask for a timestamp on each bullet? Also, can you ask for links?

**Comment by scripting** [comment]
Go ahead and try it.

**Comment by scripting** [comment]
This is where you go to try it.

https://chatthing.ai/bots/8150dc18-2086-428d-81a1-946248c381a8

**Comment by scripting** [comment]
We should figure out how to get it to read an RSS feed, with all its data, including a link to stories. 

Stripping it down to Markdown was a nice way to get started, but we're throwing out too much information. 

It doesn't have to be XML, we can convert the feed to JSON, easily.

**Comment by JPhilpin** [comment]
> Can you just ask for a timestamp on each bullet? Also, can you ask for links?

... you can ask .. I just did .? But the best you'll get back is the source ... I know that because the bot told me.

**Comment by scripting** [comment]
@JPhilpin -- i don't get what you mean. this is just a beginning project by a beginner, btw. so if you feel the need to criticize at least be learn in what you're saying. thanks.

**Comment by JPhilpin** [comment]
@scripting wrote ...

> @JPhilpin -- i don't get what you mean. this is just a beginning project by a beginner, btw. so if you feel the need to criticize at least be clear in what you're saying. thanks.

Huh? **Sorry you take it as criticism ... absolutely not** .. big fan of what you do - and this is another example of you pushing the boundaries .... the comment is a reply to a question asked earlier by @drewkime - to which you replied and said "Go ahead and try it" - so I did because I thought it was a good question ... and the response I got was something along the lines of "sorry - I am a bot and I cant provide you with URL's - only the source" - that is where the comment come from.

Interesting that just now I went to http://scripting.com/2023/04/07/205325.html?title=nytNewsbotEmbedded and asked the same question - and got an apology that it can't in fact do that ... in fact right now it is giving me the same response even to a prompt that you had as a suggestion.

This I am sure is down to the ever changing nature of bot land ... and zippo to do with what you are doing - so NOT a criticism ... 

<img width="550" alt="Xnapper-2023-04-08-09 16 15-10367140-5EC1-4751-8615-B8C8C1A0B2A8" src="https://user-images.githubusercontent.com/18621554/230731913-392ef8d6-c935-4795-afac-b4667b798999.png">


<img width="546" alt="Xnapper-2023-04-08-09 16 43-2902AD95-8AD5-40FD-B83B-DA7ADF550F47" src="https://user-images.githubusercontent.com/18621554/230731923-a967ec18-dda7-4797-bbd8-918645785530.png">

**Comment by scripting** [comment]
criticism is a _good_ thing! as long as it helps and is about the work not the person. ;-)

**Comment by jsavin** [comment]
After realizing that the bot only knows about data from the feed from April 7, I was able to get it to give me a list of headlines with timestamps, then the same list in JSON format, then the same list in JSON format with links (not tested), and finally a valid RSS 2.0 feed:

https://chatthing.ai/bots/8150dc18-2086-428d-81a1-946248c381a8/chats/05428ec1-39e6-4442-af8a-7eb07bef07a1

**Comment by jsavin** [comment]
I told it absolutely nothing about JSON or RSS except to ask for one correction to the generated RSS to make sure it finished its answer.
